### PR TITLE
Add vault v0.1.0

### PR DESCRIPTION
**Pull request by jacoelho**
Vault is a tool for securely accessing secrets. A secret is anything
that you want to tightly control access to, such as API keys, passwords,
certificates, and more. Vault provides a unified interface to any
secret, while providing tight access control and recording a detailed
audit log.

### CODE REVIEW

File: Casks/vault.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  # bintray.com is the official download host per the vendor homepage
+  url "https://dl.bintray.com/mitchellh/vault/vault_#{version}_darwin_amd64.zip"
+  name 'Vault'
+  homepage 'https://www.vaultproject.io'
+  license :oss
+
+  binary 'vault'
+end
+
```
Comment (vitorgalvao) [inline Casks/vault.rb]: This line must go.

File: Casks/vault.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+cask :v1 => 'vault' do
+  version '0.1.0'
+  sha256 'e0a1a1ace4c9577408800825b374c0239c04ebc9ab6ed18c924dc8f0393b9fd5'
+
+  # bintray.com is the official download host per the vendor homepage
+  url "https://dl.bintray.com/mitchellh/vault/vault_#{version}_darwin_amd64.zip"
+  name 'Vault'
+  homepage 'https://www.vaultproject.io'
+  license :oss
```
Comment (vitorgalvao) [inline Casks/vault.rb]: `:mpl`.

**Comment by fanquake** [comment]
Can you please squash your commits?

**Comment by jacoelho** [comment]
check again please.

**Comment by phinze** [comment]
Looks good - thank you! :grinning:

**Pull request merged**